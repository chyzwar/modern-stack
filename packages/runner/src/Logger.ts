import pc from "picocolors"

class Logger {
  public static calls: unknown[][];
 
  private readonly taskName: string;

  public constructor(taskName: string) {
    this.taskName = taskName;
  }

  public time(label: string): void {
    console.time(pc.green(`[${this.taskName}] ${label}`));
  }

  public timeEnd(label: string): void {
    console.timeEnd(pc.green(`[${this.taskName}] ${label}`));
  }

  public log(message = ""): void {
    console.log(`[${pc.green(this.taskName)}] ${message}`);
  }

  public warn(message = "", meta: unknown = ""): void {
    console.warn(pc.yellow(`[${this.taskName}] ${message}`), meta);
  }

  public error(message = "", meta: unknown = ""): void {
    console.log(`${pc.red(`[${this.taskName}]`)} ${message}`, meta);
  }

  public info(message = ""): void {
    console.log(`${pc.green(`[${this.taskName}]`)} ${message}`);
  }
}


export default Logger;