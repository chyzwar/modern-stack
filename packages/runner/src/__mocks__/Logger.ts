class Logger {
  public static calls: unknown[][] = [];

  public info(...args: unknown[]): void {
    Logger.calls.push(args);
  }

  public time(...args: unknown[]): void {
    Logger.calls.push(args);
  }

  public timeEnd(...args: unknown[]): void {
    Logger.calls.push(args);
  }

  public error(...args: unknown[]): void {
    Logger.calls.push(args);
  }
} 

export default Logger;