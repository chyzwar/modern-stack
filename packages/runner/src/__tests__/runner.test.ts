import {resolve} from "path";
import {expect, jest, describe, it} from '@jest/globals';
import Logger from "../Logger.js";

jest.mock("../Logger");

describe("runner", () => {
  it("should load configuration form runner.config.js", async() => {
    jest.spyOn(process, "cwd").mockImplementationOnce(() => {
      return resolve(__dirname, "..", "__fixtures__");
    });
    await import("../runner.js");

    expect(Logger.calls).toMatchSnapshot();
  });
});