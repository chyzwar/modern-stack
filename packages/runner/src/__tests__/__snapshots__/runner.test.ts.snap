// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runner should load configuration form runner.config.js 1`] = `Array []`;
